web/base_viewer.js

```diff
@@ -112,31 +131,38 @@ function PDFPageViewBuffer(size) {
    * `idsToKeep` has no impact on the final size of the buffer; if `idsToKeep`
    * is larger than `newSize`, some of those pages will be destroyed anyway.
    */
-  this.resize = function (newSize, idsToKeep = null) {
-    size = newSize;
+  resize(newSize, idsToKeep = null) {
+    this.#size = newSize;
+
+    const buf = this.#buf;
     if (idsToKeep) {
-      moveToEndOfArray(data, function (page) {
-        return idsToKeep.has(page.id);
-      });
+      const ii = buf.size;
+      let i = 1;
+      for (const view of buf) {
+        if (idsToKeep.has(view.id)) {
+          buf.delete(view); // Move the view to the "end" of the buffer.
+          buf.add(view);
+        }
+        if (++i > ii) {
+          break;
+        }
```

Comment:
*Please note:* The unit-test was updated to ensure that I didn't mess up this particular condition.